src/main/java/org/jvnet/hudson/test/RealJenkinsRule.java

```diff
@@ -457,6 +464,125 @@ public RealJenkinsRule prepareHomeLazily(boolean prepareHomeLazily) {
+     *   <li>-Dcom.redhat.fips=false</li>
+     * </ul>
+     * @param fipsLibrariesPath path to directory with FIPS BouncyCastle jars
+     * @return
+     */
+    public RealJenkinsRule withFIPSEnabled(@NonNull Path fipsLibrariesPath) {
+        this.fipsLibrariesPath = Objects.requireNonNull(fipsLibrariesPath);
+        // check expected BouncyCastle fips jars are here
+        if  (!Files.exists(this.fipsLibrariesPath) ||
+                !Files.exists(this.fipsLibrariesPath.resolve("bc-fips.jar")) ||
+                !Files.exists(this.fipsLibrariesPath.resolve("bctls-fips.jar")) ||
+                !Files.exists(this.fipsLibrariesPath.resolve("bcpkix-fips.jar"))) {
+            throw fipsIllegalSetupException(this.fipsLibrariesPath);
+        }
+
+        withBootClasspath(fipsLibrariesPath.resolve("bc-fips.jar").toFile(),
+                            fipsLibrariesPath.resolve("bctls-fips.jar").toFile(),
+                            fipsLibrariesPath.resolve("bcpkix-fips.jar").toFile());
+        try {
+            javaOptions("--add-exports", "java.base/sun.security.provider=ALL-UNNAMED",
+                            "-Dsecurity.overridePropertiesFile=true",
+                            "-Djava.security.properties=" + writeFIPSJavaSecurityFile().toUri(),
+                            "-Dorg.bouncycastle.fips.approved_only=true",
+                            "-Djavax.net.ssl.trustStoreType=PKCS12",
+                            "-Djenkins.security.FIPS140.COMPLIANCE=true",
+                            "-Dcom.redhat.fips=false");
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+        return this;
+    }
+
+    public RealJenkinsRule withBootClasspath(File...files) {
+        this.bootClasspathFiles.addAll(List.of(files));
+        return this;
+    }
+
+    private IllegalArgumentException fipsIllegalSetupException(Path fipsLibrariesPath) {
+        String message = "Path " + fipsLibrariesPath.toString() + " must contains the FIPS BouncyCastle jars:";
+        message += "bc-fips.jar, bctls-fips.jar and bcpkix-fips.jar" + System.lineSeparator();
+        message += "Configure your pom.xml with this plugin execution: " + System.lineSeparator();
+        message += "<plugin>\n" +
+                "  <groupId>org.apache.maven.plugins</groupId>\n" +
+                "  <artifactId>maven-dependency-plugin</artifactId>\n" +
+                "  <executions>\n" +
+                "    <execution>\n" +
+                "      <id>copy-libs</id>\n" +
+                "      <goals>\n" +
+                "        <goal>copy</goal>\n" +
+                "      </goals>\n" +
+                "      <configuration>\n" +
+                "        <artifactItems>\n" +
+                "          <artifactItem>\n" +
+                "            <groupId>org.bouncycastle</groupId>\n" +
+                "            <artifactId>bc-fips</artifactId>\n" +
+                "            <version>1.0.2.4</version>\n" +
```

Comment:
:-)   you didn't want to hard code the version in an above method, but the suggestions are hard coded here!

Reply:
it's suggestions only. users are free to test latest version of bc if they need.